packages/react/src/components/Geo/LocationSearch/index.tsx

```diff
@@ -33,18 +35,33 @@ const LocationSearchControl = ({
   return null;
 };
 
-const LocationSearchStandalone = (props: LocationSearchProps) => {
+const LocationSearchStandalone = ({
+  onLoading = () => {},
+  onResult = () => {},
+  onResults = () => {},
+  onClear = () => {},
+  onError = () => {},
+  ...props
+}: LocationSearchProps) => {
   const hasMounted = useRef(false);
 
   useEffect(() => {
     if (!hasMounted.current) {
-      (createAmplifyGeocoder(props) as unknown as AmplifyLocationSearch).addTo(
-        `#${LOCATION_SEARCH_CONTAINER}`
-      );
+      const map = createAmplifyGeocoder(
+        props
+      ) as unknown as AmplifyLocationSearch;
+
+      map.on('result', onResult);
+      map.on('loading', onLoading);
+      map.on('results', onResults);
+      map.on('clear', onClear);
+      map.on('error', onError);
+
+      map.addTo(`#${LOCATION_SEARCH_CONTAINER}`);
```

Comment:
Small question: is there a reason that we want to add the `map.on` functions before the  `map.addTo`?

Reply:
No, happy to restructure this